src/optimizer/pushdown/pushdown_projection.cpp

```diff
@@ -6,6 +6,23 @@
 
 namespace duckdb {
 
+static bool HasSideEffects(LogicalProjection &proj, const unique_ptr<Expression> &expr) {
+	if (expr->type == ExpressionType::BOUND_COLUMN_REF) {
+		auto &colref = expr->Cast<BoundColumnRefExpression>();
+		D_ASSERT(colref.binding.table_index == proj.table_index);
+		D_ASSERT(colref.binding.column_index < proj.expressions.size());
+		D_ASSERT(colref.depth == 0);
+		if (proj.expressions[colref.binding.column_index]->HasSideEffects()) {
+			return true;
+		}
+		return false;
+	}
+	bool has_side_effects = false;
+	ExpressionIterator::EnumerateChildren(
```

Comment:
I'm not sure if this is necessary as `HasSideEffects` is already recursive for expressions that have children

Reply:
Yes, you are right. I'll fix it later. thanks!

Reply:
Ah right, that makes sense, there is a layer of indirection that I was missing.

That checks out, thanks for clarifying 👍